--- a/gatb-core/CMakeLists.txt
+++ b/gatb-core/CMakeLists.txt
@@ -212,7 +212,7 @@
 set (gatb-core-includes ${PROJECT_BINARY_DIR}/include  ${PROJECT_BINARY_DIR}/include/${CMAKE_BUILD_TYPE} ${PROJECT_SOURCE_DIR}/src  ${PROJECT_SOURCE_DIR}/thirdparty ${gatb-core-extra-libraries-inc})
 
 # We define the libraries used for linking binary based on gatb core
-set (gatb-core-libraries   gatbcore-static  dl  pthread  z hdf5 ${gatb-core-extra-libraries})
+set (gatb-core-libraries   gatbcore-dynamic  dl  pthread  z hdf5 ${gatb-core-extra-libraries})
 
 # We define the directory where to find cmake helpers
 set (gatb-core-cmake  ${CMAKE_MODULE_PATH})
--- a/gatb-core/tools/CMakeLists.txt
+++ b/gatb-core/tools/CMakeLists.txt
@@ -7,7 +7,7 @@
 ################################################################################
 
 # We add the compilation options for the library
-add_definitions (${gatb-core-flags})
+add_definitions (${gatb-core-flags} -flto)
 
 include_directories (${gatb-core-includes})
 
